project_task_knowledge/models/sale_order.py

```diff
@@ -0,0 +1,15 @@
+# © 2023 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    def _action_confirm(self):
+        result = super()._action_confirm()
+        for record in self.tasks_ids:
```

Comment:
This will fail due to permissions restrictions:

![imagen](https://user-images.githubusercontent.com/12749832/223768334-3c6f05af-f207-41ba-ad30-3cb080e34376.png)

.e.g if Sales user has no access to tasks, you can check it removing a test user from both Project and Timesheets apps.

Not tested, but this fix should work:
```suggestion
        for record in self.sudo().tasks_ids:
```

In the same line, is it possible to replace `if` statement below with a `filtered` call?